ibu-imager/internal/seedcreator/seedcreator.go

```diff
@@ -183,6 +193,72 @@ func (s *SeedCreator) stopServices() error {
 	return nil
 }
 
+func (s *SeedCreator) runRecert() error {
+	s.log.Println("Running recert --force-expire tool and save a summary without sensitive data.")
+
+	// Get etcdImageRaw available for the releaseImage variable,
+	// this is needed by recert to run an unauthenticated etcd server for dry-run pre-checks.
+	etcdImage, err := getEtcdImageFromStaticDefinition()
+	if err != nil {
+		return err
+	}
+
+	// Run unauthenticated etcd server for the recert tool.
+	// This runs a small fake unauthenticated etcd server backed by the actual etcd database,
+	// which is required before running the recert tool.
+	s.log.Info("Run unauthenticated etcd server for recert dry-run")
+	_, err = s.ops.RunInHostNamespace(
+		"podman", []string{"run", "--name", "recert_etcd",
+			"--detach", "--rm", "--network=host", "--privileged", "--replace",
+			"--authfile", s.authFile, "--entrypoint", "etcd",
+			"-v", "/var/lib/etcd:/store",
+			etcdImage,
+			"--name", "editor",
+			"--data-dir", "/store"}...)
+	if err != nil {
+		return errors.Wrap(err, "Failed to run recert_etcd container")
+	}
+
+	// Ensure that the unauthenticated etcd server is killed before the runRecert
+	// function returns, even if an error occurs.
+	defer func() {
+		s.log.Info("Kill the unauthenticated etcd server")
+		_, err = s.ops.RunInHostNamespace(
+			"podman", []string{"kill", "recert_etcd"}...)
+		if err != nil {
+			s.log.Errorf("Failed to kill recert_etcd container: %v", err)
+		}
+	}()
+
+	// TODO: wait for etcd server programmatically
+	s.log.Info("Wait 10 secs for unauthenticated etcd start serving")
+	time.Sleep(10 * time.Second)
```

Comment:
Lets have smarted validation and verify etcd container is running

Reply:
Added a WaitForEtcd function to ops in [f25aa67](https://github.com/openshift-kni/lifecycle-agent/pull/35/commits/f25aa67b2014b3e84dc53fc8a579a69da52ac120).